### PR TITLE
feat: add Fordefi wallet

### PR DESCRIPTION
**Pull request by rony-arnac**
### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [ ] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### Docs Checklist
- [x] Include a screenshot of any changes ([see docs README on running locally](https://github.com/blocknative/web3-onboard/blob/develop/docs/README.md))
- [x] Add/update the appropriate package README (if applicable)
- [x] Add/update the related module in the [docs demo](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/lib/services/onboard.js) (if applicable)
- [x] Add/update the related package in the `docs/package.json` file (if applicable)

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (injected)
- [ ] send transaction
- [ ] switch chains
- [ ] sign message
- [ ] sign typed message
- [ ] disconnect

### CODE REVIEW

File: packages/injected/src/constants.ts [hunk 1]
```diff
@@ -27,5 +27,6 @@ export const WALLET_NAMES: { [key: string]: string } = {
   defiwallet: 'DeFi Wallet',
   safeheron: 'Safeheron',
   talismanEth: 'Talisman',
-  onekey: 'OneKey'
+  onekey: 'OneKey',
+  fordefi: 'Fordefi',
```
Comment (Adamj1232) [inline packages/injected/src/constants.ts:31]:
```suggestion
  fordefi: 'Fordefi'
```

Comment (rony-arnac) [reply packages/injected/src/constants.ts]: fixed!

File: packages/injected/src/types.ts [hunk 2]
```diff
@@ -138,7 +139,8 @@ export enum ProviderLabel {
   DeFiWallet = 'DeFi Wallet',
   Safeheron = 'Safeheron',
   Talisman = 'Talisman',
-  OneKey = 'OneKey'
+  OneKey = 'OneKey',
+  Fordefi = 'Fordefi',
```
Comment (Adamj1232) [inline packages/injected/src/types.ts:143]:
```suggestion
  Fordefi = 'Fordefi'
```

File: packages/injected/src/types.ts [hunk 3]
```diff
@@ -63,7 +63,8 @@ export enum ProviderIdentityFlag {
   DeFiWallet = 'isDeficonnectProvider',
   Safeheron = 'isSafeheron',
   Talisman = 'isTalisman',
-  OneKey = 'isOneKey'
+  OneKey = 'isOneKey',
+  Fordefi = 'isFordefi',
```
Comment (Adamj1232) [inline packages/injected/src/types.ts:67]:
```suggestion
  Fordefi = 'isFordefi'
```

**Comment by Adamj1232** [comment]
@rony-arnac code looks good! Wanted to check on these points again for testing prior to merge:

- Can you ensure all of the necessary testing steps are completed within the PR description for injected wallets using the internal demo app? Just run yarn && yarn dev from the root of the project
- Also, you have the ability to pass an external download link if wanted - see the MetaMask implementation for example.

There are checkboxes in the PR description to check after these have been fully tested.

**Review by Adamj1232 — APPROVED** [review]
Looks good! Thanks @rony-arnac !

**Pull request merged**